src/bin/sync-categories.rs

```diff
@@ -0,0 +1,50 @@
+// Sync available crate categories from `src/categories.txt`.
+// Only needs to be run for new databases or when `src/categories.txt`
+// has been updated.
+//
+// Usage:
+//      cargo run --bin sync-categories
+
+#![deny(warnings)]
+
+extern crate cargo_registry;
+extern crate postgres;
+
+use cargo_registry::env;
+
+fn main() {
+    let conn = postgres::Connection::connect(&env("DATABASE_URL")[..],
+                                             postgres::TlsMode::None).unwrap();
+    {
+        let tx = conn.transaction().unwrap();
+        sync(&tx).unwrap();
+        tx.set_commit();
+        tx.finish().unwrap();
+    }
+}
+
+fn sync(tx: &postgres::transaction::Transaction) -> postgres::Result<()> {
+    let categories = include_str!("../categories.txt");
+    let categories = categories.lines().collect::<Vec<_>>();
+    let insert = categories.iter()
+                           .map(|c| format!("('{}')", c))
+                           .collect::<Vec<_>>()
+                           .join(",");
```

Comment:
Itertools has [`join`](https://docs.rs/itertools/0.5.6/itertools/trait.Itertools.html#method.join) to avoid the intermediate `Vec`. If postgres allows for trailing commas, you could also do `let insert: String = ....map(|c| format!("('{}'),", c)).collect()`.